src/liquidity.rs

```diff
@@ -305,6 +310,79 @@ where
 		self.lsps2_client.as_ref().map(|s| (s.lsp_node_id, s.lsp_address.clone()))
 	}
 
+	pub(crate) fn lsps2_channel_needs_manual_broadcast(
+		&self, counterparty_node_id: PublicKey, user_channel_id: u128,
+	) -> Result<bool, APIError> {
+		// if we are not in a client_trusts_lsp model, we don't check and just return false
+		if !self.is_client_trusts_lsp() {
+			log_debug!(self.logger, "Skipping funding transaction broadcast as client trusts LSP.");
+			return Ok(false);
+		}
+
+		// if we are in a client_trusts_lsp model, then we check if the LSP has an LSPS2 operation in progress
+		self.lsps2_service.as_ref().map_or(Ok(false), |_| {
+			let lsps2_service_handler = self.liquidity_manager.lsps2_service_handler();
+			if let Some(handler) = lsps2_service_handler {
+				handler.channel_needs_manual_broadcast(user_channel_id, &counterparty_node_id)
+			} else {
+				log_error!(self.logger, "LSPS2 service handler is not available.");
+				Ok(false)
+			}
+		})
+	}
+
+	pub(crate) fn lsps2_store_funding_transaction(
```

Comment:
Given that we use them both at the same callsite and for the same purpose, should we just merge `lsps2_channel_needs_manual_broadcast` / `lsps2_store_funding_transaction` to `broadcast_or_store_funding_transaction` and call that in the event handler? Then most of this redundant boilerplate goes away, we just check once what we should do and call the respective method on channelmanager here?